### PR TITLE
Fix #93: Reject non-positive softmax temperatures

### LINKED ISSUE #93: Negative softmax_temperature silently inverts predictions; zero yields all-nan probabilities
`softmax_temperature` is used as a divisor with no guard (`tabfm/src/classifier_and_regressor.py:3372`):

```python
x = x / temperature
```

and the estimator does not define sklearn's `_parameter_constraints`, so nothing validates it on the way in either. I checked — there is no `_parameter_constraints` anywhere in the tree, which means scikit-learn's own parameter validation never runs for any of the estimator's parameters.

Reproduced on `b15593e4c1111ddb5f4f30dd2957df2edbaa04ca` against `TabFMClassifier.softmax` directly, so no checkpoint is needed:

```
temperature=0.9    -> [[0.6895, 0.227, 0.0835]]  sum=1.0000  [no warning]
temperature=0.0    -> [[nan, nan, nan]]          sum=nan     [RuntimeWarning]
temperature=-1.0   -> [[0.0961, 0.2613, 0.6426]] sum=1.0000  [no warning]
temperature=1e-300 -> [[1.0, 0.0, 0.0]]          sum=1.0000  [no warning]
```

The `-1.0` row is the one I would flag. The probabilities still sum to 1.0 and nothing warns, but the ordering is reversed — the input logits rank class 0 highest, and the output ranks class 2 highest. A negative temperature silently inverts the prediction rather than failing.

`0.0` is milder in that it at least emits a `RuntimeWarning`, but the result is all-`nan` probabilities, and a warning is easy to miss inside a fit loop.

A one-line check that `softmax_temperature > 0` would cover both. If you would rather adopt sklearn's `_parameter_constraints` for the estimator as a whole, that would cover this and the rest of the constructor arguments in one go — happy to send either.

Disclosure: I used an AI assistant to help find this. I ran the reproduction myself.

### PR DESCRIPTION
**Pull request by Echo-Nie**
Fixes https://github.com/google-research/tabfm/issues/93

Added a small check in `TabFMClassifier.softmax()` to reject `temperature <= 0` with a `ValueError`.

Also added regression tests for `0.0`, `-1.0`, and a valid positive temperature. Kept this PR scoped to #93 only.

## Tests

F2P before the fix:

```text
temperature=0.0  -> ValueError not raised
temperature=-1.0 -> ValueError not raised

pytest exit code: 1
````

Same test after the fix:

```text
1 passed, 2 subtests passed
pytest exit code: 0
```

Full new test class:

```text
2 passed, 2 subtests passed
```

Affected test file:

```text
53 passed, 22 warnings, 2 subtests passed in 43.50s
```

Manual check after the fix:

```text
temperature=0.9  -> probabilities unchanged
temperature=0.0  -> ValueError
temperature=-1.0 -> ValueError
```

@erzel @weihaokong could you take a look when you get a chance? Thx!

### CODE REVIEW

**Review by sylvesterkaczmarek — COMMENTED** [review]
One invalid temperature still slips through: `np.nan <= 0` is false, so NaN reaches `x / temperature` and the softmax returns NaNs. Since this validation is meant to enforce a valid positive temperature, could it reject non-finite values too and add a NaN case?

**Comment by Echo-Nie** [comment]
> One invalid temperature still slips through: `np.nan <= 0` is false, so NaN reaches `x / temperature` and the softmax returns NaNs. Since this validation is meant to enforce a valid positive temperature, could it reject non-finite values too and add a NaN case?

Thx for catching this! Fixed.

**Review by sylvesterkaczmarek — APPROVED** [review]
The validation now rejects all non-finite temperatures as well as values <= 0, and the regression includes NaN, both infinities, zero and negative values. This resolves the gap I raised.
